remote_attestation/rust/src/tests/message.rs

```diff
@@ -0,0 +1,222 @@
+        assert_eq!(server_identity.transcript_signature, default_array(),);
+
+        let result = test_serialize_template(&server_identity);
+        assert!(result.is_ok());
+        TestResult::from_bool(result.unwrap())
+    }
+    quickcheck(property as fn(Vec<u8>, Vec<u8>, Vec<u8>, Vec<u8>, Vec<u8>) -> TestResult);
+}
+
+#[test]
+fn test_serialize_client_identity() {
+    fn property(
+        ephemeral_public_key: Vec<u8>,
+        transcript_signature: Vec<u8>,
+        signing_public_key: Vec<u8>,
+        attestation_info: Vec<u8>,
+    ) -> TestResult {
+        if ephemeral_public_key.len() > KEY_AGREEMENT_ALGORITHM_KEY_LENGTH
+            || transcript_signature.len() > SIGNATURE_LENGTH
+            || signing_public_key.len() > SIGNING_ALGORITHM_KEY_LENGTH
+        {
+            return TestResult::discard();
+        }
+        let mut client_identity = ClientIdentity::new(
+            to_array(&ephemeral_public_key).unwrap(),
+            to_array(&signing_public_key).unwrap(),
+            attestation_info,
+        );
+        assert_eq!(client_identity.transcript_signature, default_array(),);
+
+        let transcript_signature_array = to_array(&transcript_signature).unwrap();
+        client_identity.set_transcript_signature(&transcript_signature_array);
+        assert_eq!(
+            client_identity.transcript_signature,
+            transcript_signature_array
+        );
+
+        let result = test_serialize_template(&client_identity);
+        assert!(result.is_ok());
+
+        client_identity.clear_transcript_signature();
+        assert_eq!(client_identity.transcript_signature, default_array());
+
+        TestResult::from_bool(result.unwrap())
+    }
+    quickcheck(property as fn(Vec<u8>, Vec<u8>, Vec<u8>, Vec<u8>) -> TestResult);
+}
+
+#[test]
+fn test_serialize_encrypted_data() {
+    fn property(nonce: Vec<u8>, data: Vec<u8>) -> TestResult {
+        if nonce.len() > NONCE_LENGTH {
+            return TestResult::discard();
+        }
+        let encrypted_data = EncryptedData::new(to_array(&nonce).unwrap(), data);
+        let result = test_serialize_template(&encrypted_data);
+        assert!(result.is_ok());
+        TestResult::from_bool(result.unwrap())
+    }
+    quickcheck(property as fn(Vec<u8>, Vec<u8>) -> TestResult);
+}
+
+#[test]
+fn test_deserialize_message() {
+    let client_hello = ClientHello::new(default_array());
+    let deserialized_client_hello = deserialize_message(&client_hello.serialize().unwrap());
+    assert_matches!(deserialized_client_hello, Ok(_));
+    assert_eq!(
+        deserialized_client_hello.unwrap(),
+        MessageWrapper::ClientHello(client_hello)
+    );
+
+    let server_identity =
+        ServerIdentity::new(default_array(), default_array(), default_array(), vec![]);
+    let deserialized_server_identity = deserialize_message(&server_identity.serialize().unwrap());
+    assert_matches!(deserialized_server_identity, Ok(_));
```

Comment:
Why not use `is_ok()` here? And everywhere else `assert_matches` is used.

Reply:
`is_ok()` doesn't print the error message if the test fails.

Reply:
Might be worth adding a comment, so that we don't change it later.